src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -688,7 +680,44 @@ osx_render(void *vdata,
 {
 	OSXOutput *od = (OSXOutput *) vdata;
 
-	int count = in_number_frames * od->asbd.mBytesPerFrame;
+	if (od->cancel) {
+		od->ring_buffer->reset();
+		od->cancel = false;
+	}
+
+	// If data is not available, fill the buffer with silence,
+	if (!od->ring_buffer->read_available()) {
+#ifdef ENABLE_DSD
+		if (od->dop_enabled) {
+			const auto silence = od->pcm_export->GetSilence();
+			size_t count = in_number_frames * od->asbd.mBytesPerFrame / silence.size;
+			for (size_t i = 0; i < count; ++i) {
+				memcpy((uint8_t *)buffer_list->mBuffers[0].mData + i * silence.size, silence.data, silence.size);
+			}
+			buffer_list->mBuffers[0].mDataByteSize = count * silence.size;
+			return noErr;
+		}
+#endif
+		size_t count = in_number_frames * od->asbd.mBytesPerFrame;
+		bzero(buffer_list->mBuffers[0].mData, count);
```

Comment:
I'd prefer to use `PcmSilence()` here. Even though `bzero()` happens to be correct here, doing this in a central code location is better. Right now, only DSD uses a different silence pattern; now imagine one day you add native DSD support here; using `PcmSilence()` means you don't need to refactor this one.

Reply:
Will do in future PR, together with the above comment to add silence generation back.